### PR TITLE
doc: Add release notes for #18335

### PR DESCRIPTION
**Pull request by LarryRuane**
Follow-on to #18335

### CODE REVIEW

**Comment by sipa** [comment]
Preferably put these in a `doc/release-notes-18335.md` file, so they don't get lost when release notes are aggregated for the next release.

**Comment by MarcoFalke** [comment]
@fanquake @sipa I asked the release notes to be put into the main file directly. This will save me time, because I don't have to aggregate the snippet (or anyone else doing the aggregation). IIRC the only reasons we started using snippets is:

* Makes backport easier, because snippets only end up in the backports and can easily be removed from master.
* Makes merge-conflicts less common for long-standing pull requests, because snippets don't conflict with each other.

None of the reasons apply here, because 18335 isn't tagged for backport and this pull request is not long-lived.

File: doc/release-notes.md [hunk 1]
```diff
@@ -105,6 +105,11 @@ Low-level changes
 
 RPC
 ---
+- `bitcoind` can process a limited number of simultaneous RPC requests.
+  Previously, if this limit was exceeded, `bitcoin-cli` displayed the string
```
Comment (MarcoFalke) [inline doc/release-notes.md:109]: I don't think anyone is using the cli as a stable interface. I was more referring to an http-rpc proxy receiving a different HTTP error code when the limit is exceeded.

File: doc/release-notes.md [hunk 2]
```diff
@@ -105,6 +105,10 @@ Low-level changes
 
 RPC
 ---
+- `bitcoind` can process a limited number of simultaneous RPC requests.
```
Comment (MarcoFalke) [inline doc/release-notes.md:108]:
```suggestion
- The RPC server can process a limited number of simultaneous RPC requests.
```

(You can run the RPC server via the gui as well)

**Review by maflcko — APPROVED** [review]
ACK beside nit

File: doc/release-notes.md [hunk 3]
```diff
@@ -105,6 +105,10 @@ Low-level changes
...
+  Previously, if this limit was exceeded, `bitcoind` would respond with
+  [status code 500 (`HTTP_INTERNAL_SERVER_ERROR`)](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_server_errors).
+  Now it returns the status code 503 (`HTTP_SERVICE_UNAVAILABLE`).
```
Comment (jonatack) [inline doc/release-notes.md:111]:
```suggestion
  Now it returns the status code 503 (`HTTP_SERVICE_UNAVAILABLE`). (#18335)
```

**Comment by LarryRuane** [comment]
Force-pushed review comment suggestions, thanks Jon and Marco.

**Comment by darosior** [comment]
@LarryRuane i think you messed up your push, this ended up without any commit :)

**Comment by LarryRuane** [comment]
Thanks, @darosior, I forgot to commit before push -f, should be fixed now.

**Comment by LarryRuane** [comment]
Thanks, @darosior, I forgot to commit before push -f, should be fixed now.

**Review by darosior — APPROVED** [review]
ACK 51eef4a03ee8151c1c63b902cb3254116a2a2173

**Pull request merged**